core/src/main/scala/spire/math/Polynomial.scala

```diff
@@ -541,8 +541,10 @@ trait PolynomialOverField[@sp(Double) C] extends PolynomialOverRing[C]
         val ys = y match {
           case yd: poly.PolyDense[C] => poly.PolySparse.dense2sparse(yd)
           case ys1: poly.PolySparse[C] => ys1
+          case _ => sys.error("no")
```

Comment:
I'd rather have an `unchecked` annotation at the scrutinee (same for other places where you added `sys.error`)

Reply:
the `sys.error("no")` comes because it was used on a few places before this PR. but I could also use the annotation

Reply:
That's Erik's humour then, I guess :) I'd still prefer the annotation.

Reply:
Actually the original one was `sys.error("nooooooo")` :smile: